tests/integration/tasks/tasks.spec.ts

```diff
@@ -0,0 +1,308 @@
+    await pool.end();
+  });
+
+  afterAll(async () => {
+    await prisma.$disconnect();
+  });
+
+  describe('#getTasks', function () {
+    describe('Happy Path', function () {
+      it('should return 200 status code and the matching task', async function () {
+        // create full tree
+        const job = await createJobRecord({ ...createJobRequestBody, id: faker.string.uuid() }, prisma);
+        const stage = await createStageRecord({ ...createStageWithoutTaskBody, job_id: job.id }, prisma);
+        const tasks = await createTaskRecord([{ ...createTaskBody, stage_id: stage.id }], prisma);
+
+        const response = await requestSender.getTasksByCriteria({ queryParams: { stage_id: stage.id } });
+
+        expect(response).toMatchObject({
+          status: StatusCodes.OK,
+          body: [{ id: tasks[0]!.id, stageId: stage.id }],
+        });
+      });
+
+      it('should return 200 status code and empty array', async function () {
+        const someRandomUuid = faker.string.uuid();
+        const response = await requestSender.getTasksByCriteria({ queryParams: { stage_id: someRandomUuid } });
+
+        expect(response).toSatisfyApiSpec();
+        expect(response).toMatchObject({
+          status: StatusCodes.OK,
+          body: [],
+        });
+      });
+
+      it('should return 200 status code and all the tasks if no query params were defined', async function () {
+        // create full tree
+        const job = await createJobRecord({ ...createJobRequestBody, id: faker.string.uuid() }, prisma);
+        const stage = await createStageRecord({ ...createStageWithoutTaskBody, job_id: job.id }, prisma);
+        await createTaskRecord(
+          [
+            { ...createTaskBody, stage_id: stage.id },
+            { ...createTaskBody, stage_id: stage.id },
+          ],
+          prisma
+        );
+
+        const response = await requestSender.getTasksByCriteria();
+
+        if (response.status !== StatusCodes.OK) {
+          throw new Error();
+        }
+
+        expect(response).toSatisfyApiSpec();
+        expect(response).toHaveProperty('status', StatusCodes.OK);
+        expect(response.body.length).toBeGreaterThan(0);
+      });
+
+      describe('Bad Path', function () {
+        it('should return 400 status code and a relevant validation error message when the task type is incorrect', async function () {
+          const response = await requestSender.getTasksByCriteria({ queryParams: { task_type: 'NOT_VALID_TYPE' as TaskType } });
+
+          if (response.status !== StatusCodes.BAD_REQUEST) {
+            throw new Error();
+          }
+
+          expect(response).toSatisfyApiSpec();
+          expect(response).toMatchObject({
+            status: StatusCodes.BAD_REQUEST,
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+            body: { message: expect.stringMatching(/request\/query\/task_type must be equal to one of the allowed values/) },
+          });
```

Comment:
I don't like it, it is a RECIPE FOR DISASTER. Remove the eslint comment and fix this

Reply:
This isn't for production; it's just for testing, and I think it's sufficient

Reply:
I've applied the fix to all repo, as you asked so nicely.

Reply:
Fire